runtime/lua/vim/ui.lua

```diff
@@ -9,6 +9,10 @@ local M = {}
 ---     - format_item (function item -> text)
 ---               Function to format an
 ---               individual item from `items`. Defaults to `tostring`.
+---     - type (string|nil)
+---               Type of `items`. Useful for some implentations of
+---               `select` which might inspect the structure of
+---               `items`.
```

Comment:
this explanation is useless.

and the code change just looks random. what is going on here?

Reply:
ok, maybe i did not do a good enough job explaining myself. i will try to summarize:

there are many plugins which implement lsp handlers which just replace inputlist with some other ui element from another plugin
https://github.com/gbrlsnchs/telescope-lsp-handlers.nvim, https://github.com/goolord/nvim-clap-lsp, https://github.com/weilbith/nvim-code-action-menu

these plugins *all* have to track the behavior of the codeAction handler on master sans the UI code, which i thought was unfortunate so i made this pr to separate that logic https://github.com/neovim/neovim/pull/15808

the `vim.ui.select` pr was merged afterwards. it was suggested that these plugins could just re-implement `vim.ui.,select`, which i think is the more generally useful and future-proof abstraction for this kind of plugin. 

for many would-be `vim.select.ui` replacements, like , you may want to show a preview of what's being selected. for a codeaction, this is a diff, for a workspace symbol, this is a file, etc. if in the future, `vim.ui.select` is ever used for more than one type of data (likely), you would need some mechanism to dispatch different behavior based on the structure of `input`

in the pr i linked https://github.com/neovim/neovim/pull/15808 the only person who responded to me agreed that this is a good enough solution, and it's a pretty small change, so i opened this pr :)

Reply:
I agree, for example one would like to have different ui's for code-actions and references

Reply:
I think Justin's comment was on the documentation and the commit message. You should explain the motivation for this PR in the commit message, and the docstring here should be revised to make the purpose of opt more clear. cc @mfussenegger who may have some suggestions

Reply:
Maybe it would be good put some canonical examples for type here so that all implementors of vim.ui.select. For me the obvious ones would be "file" or "uri" which could be used by a UI like telescope to show the file content or show meta information like size/permissions